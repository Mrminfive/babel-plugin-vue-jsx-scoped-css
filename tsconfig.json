{
    "compilerOptions": {
        "declaration": true,
        "lib": ["es2018"],
        "module": "commonjs",
        "moduleResolution": "node",
        "noFallthroughCasesInSwitch": true,
        "noImplicitAny": true,
        "noUnusedLocals": true,
        "noUnusedParameters": true,
        "rootDir": "./src",
        "baseUrl": "./src",
        "outDir": "./dist",
        "preserveConstEnums": false,
        "sourceMap": true,
        "strict": true,
        "target": "es2015"
    },
    "include": ["src/*.ts", "typings/*.ts"],
    "exclude": ["node_modules"]
}
