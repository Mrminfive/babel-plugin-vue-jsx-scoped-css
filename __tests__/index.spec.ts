import * as babel from '@babel/core'
import * as path from 'path'
import * as fs from 'fs'
import * as ts from 'typescript'
import * as tsconfig from 'tsconfig'
import plugin from '../dist/index'

const readFixtures = () => {
    const dir = path.resolve(__dirname, './fixtures')
    const files = fs.readdirSync(dir)

    return files.map((f) => {
        return {
            filename: f,
            path: path.resolve(dir, f)
        }
    })
}

describe('babel-plugin-vue-jsx-scoped-css', () => {
    readFixtures().forEach((fixture) => {
        it(fixture.filename, () => {
            const codeContent = fs.readFileSync(fixture.path, {
                encoding: 'utf8'
            })

            const { outputText } = ts.transpileModule(
                codeContent,
                tsconfig.loadSync(__dirname, 'tsconfig.json').config
            )

            const result = babel.transform(outputText, {
                presets: ['@vue/babel-preset-jsx'],
                plugins: [plugin],
                filename: fixture.filename
            })
            const code = result ? result.code : ''
            expect(code).toMatchSnapshot()
        })
    })
})
