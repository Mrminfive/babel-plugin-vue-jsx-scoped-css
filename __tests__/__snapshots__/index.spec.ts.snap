// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-vue-jsx-scoped-css vue.tsx 1`] = `
"import \\"./index.scoped.css?scopeId=a41f91ce\\";
import Vue from 'vue'; // @ts-ignore

import Link from './Link.vue';
export default Vue.extend({
  name: 'test-vue2',

  render() {
    const h = arguments[0];
    return h(\\"nav\\", {
      \\"class\\": 'pr-header-menu',
      \\"attrs\\": {
        \\"data-v-a41f91ce\\": \\"\\"
      }
    }, [h(\\"ul\\", {
      \\"class\\": 'menu__wrap',
      \\"attrs\\": {
        \\"data-v-a41f91ce\\": \\"\\"
      }
    }, [this.list.map(item => h(\\"li\\", {
      \\"class\\": 'menu__item',
      \\"attrs\\": {
        \\"data-v-a41f91ce\\": \\"\\"
      }
    }, [h(Link, {
      \\"attrs\\": {
        \\"href\\": item.link,
        \\"data-v-a41f91ce\\": \\"\\"
      }
    }, [item.label])]))])]);
  }

});"
`;
